internal/backend/db/dbgorm/triggers.go

```diff
@@ -125,29 +124,17 @@ func (db *gormdb) DeleteTrigger(ctx context.Context, id sdktypes.TriggerID) erro
 }
 
 func (db *gormdb) UpdateTrigger(ctx context.Context, trigger sdktypes.Trigger) error {
-	curr, err := db.GetTriggerByID(ctx, trigger.ID())
-	if err != nil {
-		return err
+	r, err := db.getTriggerByID(ctx, trigger.ID().UUIDValue())
+	if r == nil || err != nil {
+		return translateError(err)
 	}
 
-	if envID := trigger.EnvID(); envID.IsValid() && curr.EnvID() != envID {
-		return sdkerrors.ErrConflict
-	}
+	r.CodeLocation = trigger.CodeLocation().CanonicalString()
+	r.EventType = trigger.EventType()
+	r.Filter = trigger.Filter()
+	r.Schedule = trigger.Schedule()
 
-	// Note: building trigger record involves non-transactionl fetching connectionID and projectID from the DB
-	t, err := db.triggerToRecord(ctx, trigger)
-	if err != nil {
-		return err
-	}
-	// update and set null fields as well, e.g. nullify connection, data etc..
-	updateData := map[string]any{
-		"ConnectionID": t.ConnectionID,
```

Comment:
don't we need connectionID update ? you didn't move it in the refactor

Reply:
not sure i understand, but updating connection id does not make sense.